utils/grpc_health_check_provider.go

```diff
@@ -33,4 +34,5 @@ func (s *GrpcHealthServer) Watch(*he.HealthCheckRequest, he.Health_WatchServer)
 
 func RegisterHealthCheckService(s *grpc.Server) {
 	he.RegisterHealthServer(s, new(GrpcHealthServer))
+	reflection.Register(s)
```

Comment:
@JustinGaoF can you check if this is reasonable? I have only seen this line happening in main.go, not a helper.

Maybe @uchetron can chime in as well

Reply:
@thatInfrastructureGuy  Can you explain the reason to add this line? I am not understand the reason, based on my test, the L36 is enough for provide healthcheck of gRPC. 

cc @taiyangc  @uchetron  FYI, need your idea too

Reply:
From what I understand and what we use currently, reflection is an optional extension for easy discovery of services. https://github.com/grpc/grpc-go/blob/master/Documentation/server-reflection-tutorial.md

It should be done at the caller level IMO. But I intend to talk about this within the team on Thursday.

Reply:
I understand. I saw the possibility of dup code for status-server and escrow:

```
utils.RegisterHealthCheckService(gRPCServer)
reflection.Register(gRPCServer)
```
Hence, the idea was to put reflection in `utils` package. But this is definitely optional and hence, I will close this PR.